.github/workflows/ci.yml

```diff
@@ -28,16 +28,32 @@ jobs:
       - name: Install npm
         uses: actions/setup-node@v2
         with:
-          node-version: '16'
+          node-version: '18'
 
       - name: Install dependencies
         run: yarn
 
       - name: Lint
         run: npx solhint 'contracts/**/*.sol'
 
-      - name: Build
-        run: yarn build
+      - name: Install Hardhat and other dependencies
```

Comment:
If you need these dependencies, why not add them to the dev dependencies? Unusual to install selected dependencies in workflow script.